### PR TITLE
Update domain.py 

### PR DESCRIPTION
**Pull request by keiyamamo**
Two main reasons for this PR

1. There was a problem restarting the problem. The problem should be fixed with this PR. (I tested restarting and it works now)
2. `domain.py` was quite complicated to understand without enough documentation. Therefore, I added args/returns as well as underlying assumption for some of the variables that was not explicitly stated. 

Please review my comments and feel free to add comments on them!

Kei

### CODE REVIEW

File: turtleFSI/modules/domain.py [hunk 1]
```diff
@@ -7,47 +7,85 @@
...
-dx refers to the area of the domain while ds refers to the area of the boundary.
-f refere to the fluid while s refers to the solid.
+Last update: 2023-06-12
+Kei Yamamoto added docstring for assign_domain_properties function and added comments in the function.
 """
 
 def assign_domain_properties(dx, dx_f_id, rho_f, mu_f, fluid_properties, dx_s_id, material_model, rho_s, mu_s, lambda_s, solid_properties, domains, ds_s_id, boundaries, robin_bc, **namespace):
     """
     Assigns solid and fluid properties to each region.   
+
```
Comment (jorgensd) [inline turtleFSI/modules/domain.py:17]: Is there a reason for not using typing of the input arguments, instead of typing in the doc-strings?

Comment (keiyamamo) [reply turtleFSI/modules/domain.py:17]: Do you mean writing, for example, `def assign_domain_properties(dx_f_id : str or list)`?

Comment (keiyamamo) [reply turtleFSI/modules/domain.py:17]: I do not really have any reason for that. I can fix this if typing of the input arguments is preferred 👍

Comment (jorgensd) [reply turtleFSI/modules/domain.py:17]: If you do it in that way, one can use modules such as [mypy](https://mypy-lang.org/) for checking the types of input arguments.
This was a feature added in Python 3.5, see for instance: https://docs.python.org/3.8/library/typing.html

Comment (keiyamamo) [reply turtleFSI/modules/domain.py:17]: Okay, so the point of using typing of the input arguments is to automatically check the type of input arguments?

Comment (jorgensd) [reply turtleFSI/modules/domain.py:17]: Yes. It is also easy to forget to update the doc-strings everywhere, and when doing it with strings, there is no way of checking for a consistent syntax

Comment (keiyamamo) [reply turtleFSI/modules/domain.py:17]: That makes sense! Thanks for the explanation. I will update accordingly.

Comment (jorgensd) [reply turtleFSI/modules/domain.py:23]: You can be Even more specific that saying list, like
`List[dolfin.Function]`
or `Dict[key_type, value_type]`
Its up to you how detailed you'd like to be

Comment (keiyamamo) [reply turtleFSI/modules/domain.py:23]: Thanks for the comment! I think current version has enough detail for now.

**Review by dbruneau-mie — COMMENTED** [review]
Hey Kei, have you checked that these edits work with the following:
1. Single Solid Region, like the Turek Flag
2. Multiple Solid regions
3. Single Fluid region
4. Multiple Fluid regions
5. St. Venant Kirchoff material
6. Mooney Rivlin material
7. With Robin BC?
I would like to make sure this doesn't break any of the existing functionality before incorporating this PR.

**Comment by dbruneau-mie** [comment]
Hey Kei, here is a simulation with mutliple fluid regions in the Aneurysm Workflow FSI:
https://github.com/dbruneau-mie/Aneurysm_Workflow_FSI/blob/master/simulations/stenosis_FC/pulsatile_vessel.py

**Comment by dbruneau-mie** [comment]
And in the solver itself there is a cylinder with 2 solid regions
https://github.com/KVSlab/turtleFSI/blob/master/turtleFSI/problems/Test_Cylinder/problem_aneu_2solid.py

**Comment by keiyamamo** [comment]
Hi @dbruneau-mie  

Thanks for pointing out! I just reverted the change and will re-open the pull-request after I tested. 
We should consider adding multiple fluid/solid regions as a part of test...

Best,

**Comment by keiyamamo** [comment]
I did work on that, but found that the problem is multi-folded, meaning we need to fix time loop, drag/lift evaluation and not-designed-to restart. But I will work on this now that it turned out that it is important we have well designed test to avoid manually performing many kinds of test.

**Pull request merged**